src/H5Fsuper.c

```diff
@@ -361,19 +361,13 @@ H5F__super_read(H5F_t *f, H5P_genplist_t *fa_plist, bool initial_read)
 
         /* Search for the file's signature only with rank 0 process */
         if (0 == mpi_rank) {
-            herr_t status;
-
             /* Try detecting file's signature */
             /* (Don't leave before Bcast, to avoid hang on error) */
-            H5E_BEGIN_TRY
+            H5E_PAUSE_ERRORS
             {
-                status = H5FD_locate_signature(file, &super_addr);
+                H5FD_locate_signature(file, &super_addr);
```

Comment:
This should be fine, but interesting note: I looked at H5FD_locate_signature to make sure it won't update sig_addr on failure, and it looks like it's actually meant to set it to HADDR_UNDEF on failure, but it does so *after* the call to HGOTO_ERROR...

Reply:
🙃